src/VirtoCommerce.SimpleExportImportModule.Data/Services/CsvPriceDataValidator.cs

```diff
@@ -0,0 +1,128 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using CsvHelper;
+using VirtoCommerce.Platform.Core.Assets;
+using VirtoCommerce.SimpleExportImportModule.Core;
+using VirtoCommerce.SimpleExportImportModule.Core.Models;
+using VirtoCommerce.SimpleExportImportModule.Core.Services;
+using VirtoCommerce.SimpleExportImportModule.Data.Models;
+
+namespace VirtoCommerce.SimpleExportImportModule.Data.Services
+{
+    public sealed class CsvPriceDataValidator : ICsvPriceDataValidator
+    {
+        private readonly IBlobStorageProvider _blobStorageProvider;
+
+        public CsvPriceDataValidator(IBlobStorageProvider blobStorageProvider)
+        {
+            _blobStorageProvider = blobStorageProvider;
+        }
+        public async Task<ImportDataValidationResult> ValidateAsync(string fileUrl)
+        {
+            var errorsList = new List<string>();
+
+            var blobInfo = await _blobStorageProvider.GetBlobInfoAsync(fileUrl);
+
+            if (blobInfo == null)
+            {
+                errorsList.Add(ModuleConstants.ValidationErrors.FileNotExisted);
+            }
+            else if (blobInfo.Size > ModuleConstants.Settings.FileMaxSize)
+            {
+                errorsList.Add(ModuleConstants.ValidationErrors.ExceedingFileMaxSize);
+            }
+            else
+            {
+                await using var stream = _blobStorageProvider.OpenRead(fileUrl);
+                var csvConfiguration = new ImportConfiguration();
+                using var streamReader = new StreamReader(stream);
+                using var csvReader = new CsvReader(streamReader, csvConfiguration);
+
+                await ValidateDelimiterAndDataExists(streamReader, csvConfiguration.Delimiter, errorsList);
+
+                ValidateRequiredColumns(streamReader, csvReader, errorsList);
+
+                ValidateLineLimit(csvReader, errorsList);
+            }
+
+            var result = new ImportDataValidationResult { Errors = errorsList.ToArray() };
+
+            return result;
+        }
+
+        private static void ValidateLineLimit(CsvReader csvReader, List<string> errorsList)
+        {
+            if (errorsList.Count == 0)
+            {
+                ReadCsvHeader(csvReader);
+
+                var totalCount = 1;
```

Comment:
Should 0. Try to remove `+ 1` from test for exceeding lines and see what happen. It also mean we should write tests not only for invalid csv ("positive scenarios"), but also for valid csv ("negative scenarios")